app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -110,6 +114,12 @@ open class HomeActivity : LocaleAwareAppCompatActivity() {
                 ?.also { components.analytics.metrics.track(Event.OpenedApp(it)) }
         }
         supportActionBar?.hide()
+
+        // Just before the root container is drawn for the first time, consider Fenix to be visually
+        // complete.
+        rootContainer.doOnPreDraw {
+            this.asContext.application.onApplicationVisuallyComplete()
```

Comment:
There are two issues with the current implementation:
- this is not visual completeness: we're attaching a pre-draw listener in `HomeActivity.onCreate`, before any other frames are drawn, so the first onPreDraw call will be for the first frame drawn
- If we don't launch `HomeActivity` (e.g. the framework has removed us from memory and we reopen the app on the Settings screen), our visually complete jobs will never run. We need to be resilient to the app being opened from any entry point

fwiw, solving these problems is the real challenge of this issue.